integration-tests/src/nft_marketplace.rs

```diff
@@ -135,3 +137,102 @@ fn create_class_should_fail_when_relay_chain_location_not_registered() {
+#[test_case(ClassType::Auction, NOT_ALLOW ; "auction class")]
+#[test_case(ClassType::HydraHeads, NOT_ALLOW ; "hydra heads class")]
+fn test_nft_permission_for_minting(class_type: ClassType, is_allowed: bool) {
+	assert_eq!(
+		is_allowed,
+		<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_mint(&class_type)
+	);
+}
+
+#[test_case(ClassType::Marketplace, ALLOW ; "marketplace class")]
+#[test_case(ClassType::LiquidityMining, ALLOW ; "liquidity mining class")]
+#[test_case(ClassType::Redeemable, NOT_ALLOW ; "redeemable class")]
+#[test_case(ClassType::Auction, NOT_ALLOW ; "auction class")]
+#[test_case(ClassType::HydraHeads, NOT_ALLOW ; "hydra heads class")]
+fn test_nft_permission_for_transfer(class_type: ClassType, is_allowed: bool) {
+	assert_eq!(
+		is_allowed,
+		<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_transfer(&class_type)
+	);
+}
+
+#[test_case(ClassType::Marketplace, ALLOW ; "marketplace class")]
+#[test_case(ClassType::LiquidityMining, NOT_ALLOW ; "liquidity mining class")]
+#[test_case(ClassType::Redeemable, NOT_ALLOW ; "redeemable class")]
+#[test_case(ClassType::Auction, NOT_ALLOW ; "auction class")]
+#[test_case(ClassType::HydraHeads, NOT_ALLOW ; "hydra heads class")]
+fn test_nft_permission_for_burning(class_type: ClassType, is_allowed: bool) {
+	assert_eq!(
+		is_allowed,
+		<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_burn(&class_type)
+	);
+}
+
+#[test_case(ClassType::Marketplace, ALLOW ; "marketplace class")]
+#[test_case(ClassType::LiquidityMining, NOT_ALLOW ; "liquidity mining class")]
+#[test_case(ClassType::Redeemable, NOT_ALLOW ; "redeemable class")]
+#[test_case(ClassType::Auction, NOT_ALLOW ; "auction class")]
+#[test_case(ClassType::HydraHeads, NOT_ALLOW ; "hydra heads class")]
+fn test_nft_permission_for_class_destroying(class_type: ClassType, is_allowed: bool) {
+	assert_eq!(
+		is_allowed,
+		<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_destroy(&class_type)
+	);
+}
+
+#[test_case(ClassType::Marketplace, ALLOW ; "marketplace class")]
+#[test_case(ClassType::LiquidityMining, NOT_ALLOW ; "liquidity mining class")]
+#[test_case(ClassType::Redeemable, NOT_ALLOW ; "redeemable class")]
+#[test_case(ClassType::Auction, NOT_ALLOW ; "auction class")]
+#[test_case(ClassType::HydraHeads, NOT_ALLOW ; "hydra heads class")]
+fn test_nft_permission_for_deposit(class_type: ClassType, is_allowed: bool) {
+	assert_eq!(
+		is_allowed,
+		<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::has_deposit(&class_type)
+	);
+}
+
+#[test_case(ClassType::Marketplace; "marketplace class")]
+#[test_case(ClassType::LiquidityMining; "liquidity mining class")]
+#[test_case(ClassType::Redeemable; "redeemable class")]
+#[test_case(ClassType::Auction; "auction class")]
+#[test_case(ClassType::HydraHeads; "hydra heads class")]
+fn create_class_should_be_without_deposit_when_created_using_trait(class_type: ClassType) {
```

Comment:
I would like to have tests like this also for the rest of the trait’s functions.

Reply:
Ok. But in this case we are not talking about integration tests because we are just testing one implementation of the traits and there is no interaction with other pallets involved. And testing the implementation of it implies that it will also work when integrated into other pallets. So I would add tests for this directly to NFT pallet in warehouse repo. @dmoka what do you think?

Reply:
In my opinion this should be in the basilisk repo in the nft integration test (not in the marketplace). 
e.g if someone removes classType or changes permissions in the warehouse and updates tests in the warehouse, we may not catch it. It's the same situations as we had with minting without deposits, warehouse repo was updated and merged and we found problem after we tried to use NFT pallet in LM.

In my opinion this is integration test because it test integration of NFT pallet in basilisk chain with basilisk specific requirements.

Reply:
The generic core logic that we are testing here is basically this:

https://github.com/galacticcouncil/warehouse/blob/85257f53b0da5827c103bf4f739474cc1e83d3f7/nft/src/types.rs#L75

But we are actually testing how this pallet_nft is configured/integrated within Basilisk runtime, because we run this:

Because we are calling `<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::...`

... with `ClassType` enum used in the basilisk runtime pallet_nft config implementation

So I think this is fine as an integration test.

Reply:
Just reading @martinfridrich's answer, I agree with that, as it is not only related to marketplace, we could put them into a separate integration test file.

Reply:
Hmm. Ok. But I would prefer to have tests next to implementations. Right now we use structs and trait implementations from warehouse crate and test them in Basilisk repo. And we will need to do the same in Hydra repo. In my opinion the best solution would be to not use generic names for structs and traits (e.g. CreateTypedClassWithoutPermissionCheck). Then we don't need to be afraid of some changes in warehouse repo. But I'm totally fine with what you agreed on.